code/modules/research/designs/weapon_designs.dm

```diff
@@ -127,6 +127,42 @@
 	build_path = /obj/item/weapon/grenade/chem_grenade/large
 	category = list("Weapons")
 
+/datum/design/pyro_grenade
+	name = "Pyro Grenade"
+	desc = "An advanced grenade that is able to self ignite its mixture."
+	id = "pyro_Grenade"
+	req_tech = list("combat" = 3, "materials" = 3)
+	build_type = PROTOLATHE
+	materials = list(MAT_METAL = 2000, MAT_PLASMA = 500)
+	reliability = 79
+	build_path = /obj/item/weapon/grenade/chem_grenade/pyro
+	category = list("Weapons")
+
+/* // Currently commented out, because it has no worthwhile useage yet.
+
+/datum/design/cryo_grenade
+	name = "Cryo Grenade"
+	desc = "An advanced grenade that rapidly cools its contents upon detonation."
+	id = "cryo_Grenade"
+	req_tech = list("combat" = 3, "materials" = 3)
+	build_type = PROTOLATHE
+	materials = list(MAT_METAL = 2000, MAT_SILVER = 500)
+	reliability = 79
+	build_path = /obj/item/weapon/grenade/chem_grenade/cryo
+	category = list("Weapons")
+*/
+
+/datum/design/adv_grenade
+	name = "Advanced Release Grenade"
```

Comment:
The cryo im thinking would be for stuff like freezing wet floors into ice, or frostoil stuff, etc. idk man. It's just intended as a compliment to the pyro.

I tested a flashbang 'advanced release bomb' with 4 bluespace beakers, and it's a pretty effective area denial device. If you enter the radius, you're flashed until you go blind.